### PR TITLE
Add API to expose distributed reward ratio by delegate name

### PR DESCRIPTION
**Pull request by Bharat2704**
- Updated graphql schema to support distributionRatio parameter
- Added logic to resolver to add ratioList for the delegate
- Added service method to fetch rewaards ratio values for a delegate

### CODE REVIEW

File: queryprotocol/hermes2/protocol.go [hunk 1]
```diff
@@ -21,15 +22,16 @@ const (
...
+	fromTable            				   = "FROM %s"
 	selectVoter                            = "SELECT `to`, from_epoch, to_epoch, amount, t1.action_hash, `timestamp` "
 	delegateFilter                         = "WHERE delegate_name = ? "
 	selectHermesDistributionByDelegateName = selectVoter + fromJoinedTables + delegateFilter + timeOrdering
 
 	selectDelegate                         = "SELECT delegate_name, from_epoch, to_epoch, amount, t1.action_hash, `timestamp` "
 	voterFilter                            = "WHERE `to` = ? "
-	selectHermesDistributionByVoterAddress = selectDelegate + fromJoinedTables + voterFilter + timeOrdering
-
+	selectHermesDistributionByVoterAddress = selectDelegate + fromTable + voterFilter + timeOrdering
```
Comment (lizhefeng) [inline queryprotocol/hermes2/protocol.go:32]: Why do you change fromJoinedTables to fromTable?

File: queryprotocol/hermes2/protocol.go [hunk 2]
```diff
@@ -21,15 +22,16 @@ const (
...
+	selectDistributionRatio                = "SELECT block_reward_percentage, epoch_reward_percentage, foundation_bonus_percentage"
+	selectDistributionRatioByDelegateName  = selectDistributionRatio + fromTable + delegateFilter + timeOrdering
```
Comment (lizhefeng) [inline queryprotocol/hermes2/protocol.go:34]: Ditto.

File: queryprotocol/hermes2/protocol.go (hunk 1 above)
Comment (Bharat2704) [reply queryprotocol/hermes2/protocol.go:32]: I was intended to change `selectDistributionRatioByDelegateName` earlier. I think it got changed in pasting. Would correct it and push the change.

File: queryprotocol/hermes2/protocol.go (hunk 2 above)
Comment (Bharat2704) [reply queryprotocol/hermes2/protocol.go:34]: I wasn't sure about using the two joined tables. Let me know if I can use the same `fromJoinedTables` with tables `votings.VotingResultTableName` and `actions.HermesContractTableName`

Comment (lizhefeng) [reply queryprotocol/hermes2/protocol.go:34]: Sorry this one is fine. We don't need to use fromJoinedTables indeed.

File: graphql/resolver.go [hunk 3]
```diff
@@ -1602,6 +1602,25 @@ func (r *queryResolver) getHermes2ByDelegate(ctx context.Context, startEpoch int
 			voterInfoList = append(voterInfoList, info)
 		}
 	}
+	distributionRatioList := make([]*DistributionRatioInfo, 0)
+	if haveField(ctx, "byDelegate", "distributionRatio") {
+		res, err := r.HP.GetHermes2Ratio(harg, delegateName)
```
Comment (lizhefeng) [inline graphql/resolver.go:1607]: the delegateName in voting_result table is encoded. So you need to first use EncodeDelegateName() to encode the name and then pass into GetHermes2Ratio().

**Comment by Bharat2704** [comment]
Closing this pull request as per recent discussion with @lizhefeng . Would make the file changes in correct order and raise another PR from a new branch.

**Pull request closed without merging**